lib/widgets/content.dart

```diff
@@ -1196,6 +1203,61 @@ class GlobalTime extends StatelessWidget {
   }
 }
 
+class MessageTable extends StatelessWidget {
+  const MessageTable({super.key, required this.node});
+
+  final TableNode node;
+
+  @override
+  Widget build(BuildContext context) {
+    return SingleChildScrollViewWithScrollbar(
+      scrollDirection: Axis.horizontal,
+      child: Padding(
+        padding: const EdgeInsets.symmetric(horizontal: 5),
+        child: Table(
+          border: TableBorder.all(
+            color: const HSLColor.fromAHSL(1, 0, 0, 0.33).toColor()),
+          defaultColumnWidth: const IntrinsicColumnWidth(),
+          children: node.rows.map((row) {
+            return TableRow(
+              decoration: row.isHeading
+                ? const BoxDecoration(color: Colors.black)
+                : null,
+              children: row.cells.map((cell) {
+                return MessageTableCell(node: cell, isHeading: row.isHeading);
+              }).toList(growable: false),
+            );
+          }).toList(growable: false),
+        ),
+      ),
+    );
+  }
+}
+
+class MessageTableCell extends StatelessWidget {
+  const MessageTableCell({super.key, required this.node, required this.isHeading});
+
+  final TableCellNode node;
+  final bool isHeading;
+
+  @override
+  Widget build(BuildContext context) {
+    return TableCell(
+      verticalAlignment: TableCellVerticalAlignment.middle,
+      child: node.nodes.isNotEmpty
+        ? Padding(
+            padding: const EdgeInsets.all(5),
+            child: _buildBlockInlineContainer(
+              node: node,
+              style: isHeading
+                ? DefaultTextStyle.of(context).style
+                    .merge(weightVariableTextStyle(context, wght: 700))
+                : DefaultTextStyle.of(context).style,
+              textAlign: node.textAlign))
+        : const SizedBox.shrink());
```

Comment:
Should we include the `Padding` in this case too? This looks like it could make a cell that doesn't take up any space at all, and I don't think that's consistent with the web app. Here's what the web app does with a table that's nothing but empty cells:

https://chat.zulip.org/#narrow/channel/7-test-here/topic/Chris/near/1971419
<img width="806" alt="image" src="https://github.com/user-attachments/assets/56d846a3-bf2a-4b83-b5b3-09c90f2aee4b">

and each one does take up some space, and from the inspector it looks like that's coming from some padding (though 4px in all directions instead of 5px).